hack/logcheck.conf

```diff
@@ -0,0 +1,30 @@
+# This file contains regular expressions that are matched against <pkg>/<file>,
+# for example k8s.io/cmd/kube-scheduler/app/config/config.go.
+#
+# By default, structured logging call parameters are checked, but usage of
+# those calls is not required. That is changed on a per-file basis.
+#
+# Remember to clean the golangci-lint cache when changing the configuration and
+# running the verify-golangci-lint.sh script multiple times, otherwise
+# golangci-lint will report stale results:
+#    _output/local/bin/golangci-lint cache clean
+
+# At this point we don't enforce the usage structured logging calls except in
+# those packages that were migrated. This disables the check for other files.
+-structured *
+
+# Now enable it again for migrated packages.
+structured k8s.io/kubernetes/cmd/kube-proxy/.*
+structured k8s.io/kubernetes/cmd/kube-scheduler/.*
+structured k8s.io/kubernetes/cmd/kubelet/.*
+structured k8s.io/kubernetes/pkg/kubelet/.*
+structured k8s.io/kubernetes/pkg/proxy/.*
+structured k8s.io/kubernetes/pkg/scheduler/apis/.*
+structured k8s.io/kubernetes/pkg/scheduler/framework/.*
+structured k8s.io/kubernetes/pkg/scheduler/internal/cache/fake/.*
+structured k8s.io/kubernetes/pkg/scheduler/internal/heap/.*
+structured k8s.io/kubernetes/pkg/scheduler/internal/queue/.*
+structured k8s.io/kubernetes/pkg/scheduler/metrics/.*
+structured k8s.io/kubernetes/pkg/scheduler/profile/.*
+structured k8s.io/kubernetes/pkg/scheduler/testing/.*
+structured k8s.io/kubernetes/pkg/scheduler/util/.*
```

Comment:
Could we mark parent directory as migrated instead? We could not do it previously due to limitations of the logcheck before.

Reply:
You are right, a single line is enough now.

Reply:
Argh, I fell into that trap that is mentioned in the file (stale golangci-lint cache). After cleaning the cache, logcheck (correctly) points out [the remaining work](https://github.com/kubernetes/kubernetes/issues/105841):
```
pkg/scheduler/internal/cache/debugger/dumper.go:49:3: unstructured logging function "Info" should not be used (logcheck)
		klog.Info(d.printNodeInfo(name, nodeInfo))
		^
pkg/scheduler/internal/cache/debugger/dumper.go:60:2: unstructured logging function "Infof" should not be used (logcheck)
	klog.Infof("Dump of scheduling queue:\n%s", podData.String())
```

I think we concluded that a text dump is okay here, also for JSON. I'll include a conversion of these lines to `klog.InfoS`.

Reply:
Done.

The current kube-scheduler output with these changes is in the updated commit message of the first commit.